events/validators.go

```diff
@@ -51,24 +83,32 @@ func ValidateEvent(name string, t EventType, got []byte) error {
 	case CloudEvent:
 		return validateCloudEvent(name, got, want)
 	}
+
+	// Should be unreachable.
 	return nil
 }
 
-func validateLegacyEvent(name string, gotBytes, wantBytes []byte) error {
+func validateLegacyEvent(name string, gotBytes, wantBytes []byte) *ValidationInfo {
+	vi := &ValidationInfo{
+		Name: name,
+	}
 	got := make(map[string]interface{})
 	err := json.Unmarshal(gotBytes, &got)
 	if err != nil {
-		return fmt.Errorf("unmarshalling function-received version of legacy event %q: %v", name, err)
+		vi.Errs = []error{fmt.Errorf("unmarshalling function-received version of legacy event %q: %v", name, err)}
+		return vi
 	}
 
 	want := make(map[string]interface{})
 	err = json.Unmarshal(wantBytes, &want)
 	if err != nil {
-		return fmt.Errorf("unmarshalling expected contents of legacy event %q: %v", name, err)
+		vi.Errs = []error{fmt.Errorf("unmarshalling expected contents of legacy event %q: %v", name, err)}
+		return vi
 	}
 
 	if !reflect.DeepEqual(got["data"], want["data"]) {
-		return fmt.Errorf("unexpected data in event %q:\ngot %v,\nwant %v", name, got["data"], want["data"])
+		vi.Errs = []error{fmt.Errorf("unexpected data in event %q:\ngot %v,\nwant %v", name, got["data"], want["data"])}
```

Comment:
I think we should append the error instead of return because we can validate `fields` independent of `data`.